### PR TITLE
Add config options for self-chill builds

### PR DESCRIPTION
**Pull request by Lilylicious**
Adds an input field for base chill effect to the config. This is then modified by self chill effect.

Also added a few ModParser lines to catch more "ailment while X" mods.

This should correct self-chilling PoBs.

Previous behaviour with self-chill and the chilled option enabled in config:
Action speed increased by 30 * inc chill effect on self

New behaviour which matches the wiki:
Action speed increased by 10 (or player input value) * inc chill effect (on self (and against target, if self chilled)), capped at 30

### CODE REVIEW

**Comment by Lilylicious** [comment]
> Is this also in effect when you have the Elementalist's Shaper of Winter node, which makes max chill 40%

Not currently. There is some conflicting with #3271, and I don't want to duplicate code too much. Once #3721 is merged we can use their ChillMax in place of the 30.

**Comment by Lilylicious** [comment]
This should be good to merge.

File: src/Modules/ConfigOptions.lua [hunk 1]
```diff
@@ -726,12 +727,20 @@ return {
...
+		modList:NewMod("Condition:Burning", "FLAG", true, "Config", { type = "Condition", var = "Combat" })
 	end },
 	{ var = "conditionChilled", type = "check", label = "Are you Chilled?", ifCond = "Chilled", apply = function(val, modList, enemyModList)
 		modList:NewMod("Condition:Chilled", "FLAG", true, "Config", { type = "Condition", var = "Combat" })
 	end },
+	{ var = "conditionChilledEffect", type = "count", label = "Effect of Chill:", ifOption = "conditionChilled", apply = function(val, modList, enemyModList)
+		modList:NewMod("ChillVal", "OVERRIDE", val, "Chill", { type = "Condition", var = "Chilled" })
+	end },
+	{ var = "conditionSelfChill", type = "check", label = "Did you chill yourself?", ifOption = "conditionChilled", apply = function(val, modList, enemyModList)
+		modList:NewMod("Condition:ChilledSelf", "FLAG", true, "Config", { type = "Condition", var = "Combat" })
```
Comment (Wires77) [inline src/Modules/ConfigOptions.lua:739]: If we're providing an override anyway, do we really need this extra checkbox?

**Review by Wires77 — CHANGES_REQUESTED** [review]
Any config value that has `implyCond` doesn't need that condition also added to the mods it grants.

File: src/Modules/ConfigOptions.lua (hunk 1 above)
Comment (Lilylicious) [reply src/Modules/ConfigOptions.lua:739]: The override only says how much you're chilled for. When you chill yourself, effects that are normally applied on the enemy (increased effect of chill) now apply to you. We need to specify whether you caused the chill or if an enemy/effect did.

**Review by Wires77 — APPROVED** [review]
LGTM

**Pull request merged**